spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Adagrad.scala

```diff
@@ -25,36 +25,32 @@ import scala.reflect.ClassTag
 /**
  * An implementation of Adagrad. See the original paper:
  * http://jmlr.org/papers/volume12/duchi11a/duchi11a.pdf
- * @param ev numeric operator
- * @tparam T numeric type
+ * @param learningRate learning rate
+ * @param learningRateDecay learning rate decay
+ * @param weightDecay weight decay
+ * @tparam T
  */
-class Adagrad[@specialized(Float, Double) T: ClassTag](implicit ev: TensorNumeric[T])
-  extends OptimMethod[T] {
+class Adagrad[@specialized(Float, Double) T: ClassTag](
+    var learningRate: Double = 1e-3,
+    var learningRateDecay: Double = 0.0,
+    var weightDecay: Double = 0.0
+  )(implicit ev: TensorNumeric[T]) extends OptimMethod[T] {
 
   /**
    * Adagrad implementation for Adagrad
    *
    * @param feval     a function that takes a single input (X), the point of a evaluation, and
    *                  returns f(X) and df/dX
    * @param parameter the initial point
-   * @param config    a table with configuration parameters for the optimizer
-   *                  config("learningRate") : learning rate
-   *                  config("learningRateDecay") : learning rate decay
-   * @param state     a table describing the state of the optimizer; after each call the state
```

Comment:
State is used to pass parameters to outside or used for snapshot, right? 
any documentations or rules which parameters are included in state?
Will it be a problem, if in training, a user uses one kind of optimMethod, e.g. SGD, and pauses, then resume the training using another kind of optimMethod, e.g. LBFGS?